distributed/shuffle/_shuffle_extension.py

```diff
@@ -455,9 +523,21 @@ async def _get_shuffle(
                 return self.shuffles[shuffle_id]
 
     async def close(self) -> None:
+        if self.closed:
+            await self._closed_event.wait()
+            return
```

Comment:
How can it happen that _the extension_ is closed multiple times concurrently? The worker closing should be protected from this and nothing else is calling close, is it?

Reply:
It shouldn't happen during _normal_ operations. I've triggered it at some point doing funny business during testing. While I was making `Shuffle.close` idempotent tackling this method as well didn't feel like a lot of added complexity. I'm happy to drop it though.

Reply:
Replacing with an assertion to catch situations where this happens through manual intervention.